Zend/zend_compile.c

```diff
@@ -3911,16 +3921,67 @@ static bool zend_compile_function_name(znode *name_node, zend_ast *name_ast) /*
 }
 /* }}} */
 
-static void zend_compile_ns_call(znode *result, znode *name_node, zend_ast *args_ast, uint32_t lineno) /* {{{ */
+// FIXME: Move up implementation
+
+static uint32_t zend_compile_frameless_icall_ex(znode *result, zend_ast_list *args, zend_function *fbc, const zend_frameless_function_info *frameless_function_info, uint32_t type);
+static bool zend_args_contain_unpack_or_named(zend_ast_list *args);
+static const zend_frameless_function_info *find_frameless_function_info(zend_ast_list *args, zend_function *fbc, uint32_t type);
+
+static void zend_compile_ns_call(znode *result, znode *name_node, zend_ast *args_ast, uint32_t lineno, uint32_t type) /* {{{ */
 {
+	int name_constants = zend_add_ns_func_name_literal(Z_STR(name_node->u.constant));
+
+	/* Find frameless function with same name */
+	zend_function *frameless_function = NULL;
+	if (args_ast->kind != ZEND_AST_CALLABLE_CONVERT
+	 && !zend_args_contain_unpack_or_named(zend_ast_get_list(args_ast))) {
+		zend_function **flf_ptr = zend_flf_functions;
+		while (*flf_ptr != NULL) {
```

Comment:
Linear search looks like a bad idea.
Can we do ``zend_hash_find(EG(function_table), lc_func_name)`` and then check some flags?

Reply:
Not sure why I did it like that in the first place.